app/models/assignment.rb

```diff
@@ -723,10 +723,22 @@ def criterion_class
   end
 
   def get_criteria
+    is_peer_review? ? get_peer_visible_criteria : get_ta_visible_criteria
```

Comment:
The `is_peer_review` check is not correct. Add a parameter which takes a symbol, either `:all`, `:ta`, and `:peer`, and returns the correct criteria in each of these three cases.